src/main/java/co/inventorsoft/academy/springboottask/EmailService.java

```diff
@@ -16,32 +15,38 @@
 	private JavaMailSender emailSender;
 	private Logger log;
 	private EmailDAO emailDAO;
+	private ThreadPoolTaskScheduler scheduler;
 	
 	@Autowired
-	public EmailService(JavaMailSender emailSender, Logger log, @Qualifier("emailFile") EmailDAO emailDAO) {
+	public EmailService(JavaMailSender emailSender, Logger log, EmailDAO emailDAO, ThreadPoolTaskScheduler scheduler) {
 		this.emailSender = emailSender;
 		this.log = log;
 		this.emailDAO = emailDAO;
+		this.scheduler = scheduler;
 	}
 
-	@Async
 	public void sendFutureEmail() {
-		SimpleMailMessage email = emailDAO.getEmail();
-		emailDAO.saveEmail();
-		try{
-			long milliesToWait = email.getSentDate().getTime() - (new Date()).getTime();
-			if(milliesToWait > 0) {
-				log.warn(" Waiting... {} ", email.getSentDate());
-				Thread.sleep(milliesToWait);
-			}
-			log.info("\n E-mail \n To: {} \n Subject: {} \n Date: {} ", email.getTo(), email.getSubject(), email.getSentDate());
-			emailSender.send(email);
-			emailDAO.clearEmail();
-		}
-		catch(Exception e) {
-			log.warn(" E-mail was not sent ");
-			log.error(e.getMessage());
+		List<SimpleMailMessage> emails = emailDAO.getAll();
+        scheduler.setPoolSize(2);
```

Comment:
This is initialization attributes of your dependency. You shouldn't change it here. Instead initialize it properly https://github.com/inventorsoft-academy/spring-boot-task/pull/1/commits/4532772fae5e12dbd8ed4bef79563c8be951ea6b#diff-26311d95a11ece1cb2f7689c8561f1c3R54